swap/src/monero/wallet.rs

```diff
@@ -1,38 +1,45 @@
 use crate::monero::{
-    Amount, CreateWallet, CreateWalletForOutput, InsufficientFunds, OpenWallet, PrivateViewKey,
-    PublicViewKey, Transfer, TransferProof, TxHash, WatchForTransfer,
+    Amount, CreateWallet, CreateWalletForOutput, CreateWalletForOutputThenLoadDefaultWallet,
+    GetAddress, InsufficientFunds, OpenWallet, PrivateViewKey, PublicViewKey, Refresh, Transfer,
+    TransferProof, TxHash, WalletBlockHeight, WatchForTransfer,
 };
 use ::monero::{Address, Network, PrivateKey, PublicKey};
 use anyhow::Result;
 use async_trait::async_trait;
 use backoff::{backoff::Constant as ConstantBackoff, future::retry};
 use bitcoin::hashes::core::sync::atomic::AtomicU32;
-use monero_rpc::wallet;
+use monero_rpc::{
+    wallet,
+    wallet::{BlockHeight, Refreshed},
+};
 use std::{
     str::FromStr,
     sync::{atomic::Ordering, Arc},
     time::Duration,
 };
+use tokio::sync::Mutex;
 use tracing::info;
 use url::Url;
 
 #[derive(Debug)]
 pub struct Wallet {
-    pub inner: wallet::Client,
+    pub inner: Mutex<wallet::Client>,
```

Comment:
These should all be private.

Reply:
I totally agree and actually had made them private, but then I would have to add a separate constructor for the tests. Maybe more generally asked: How shall we deal with the wallet in the tests? - I find it weird that we have to "make things public" for the test...

Reply:
https://github.com/comit-network/xmr-btc-swap/pull/225/commits/578d23d7fc9d2f703e96384a2b3e911aaec176a6